src/main/java/seedu/address/logic/commands/CopyCommand.java

```diff
@@ -0,0 +1,180 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.model.person.fields.Address.PREFIX_ADDRESS;
+import static seedu.address.model.person.fields.Assets.PREFIX_ASSET;
+import static seedu.address.model.person.fields.Email.PREFIX_EMAIL;
+import static seedu.address.model.person.fields.Name.PREFIX_NAME;
+import static seedu.address.model.person.fields.Phone.PREFIX_PHONE;
+import static seedu.address.model.person.fields.Tags.PREFIX_TAG;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.util.ArgumentMultimap;
+import seedu.address.logic.util.ArgumentTokenizer;
+import seedu.address.logic.util.ParserUtil;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Copies the details of a person in the address book.
+ */
+public class CopyCommand extends Command {
+
+    public static final String COMMAND_WORD = "copy";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Copies the details of the prefix identified\n"
+        + "Parameters: "
+        + "[" + PREFIX_NAME + "] "
+        + "[" + PREFIX_PHONE + "] "
+        + "[" + PREFIX_EMAIL + "] "
+        + "[" + PREFIX_ADDRESS + "] "
+        + "[" + PREFIX_TAG + "] "
+        + "[" + PREFIX_ASSET + "]\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_NAME + " "
+        + PREFIX_PHONE + " "
+        + PREFIX_TAG + " "
+        + PREFIX_ASSET + " ";
+
+    public static final String MESSAGE_NO_PARAM = "At least one field to copy must be provided.";
+
+    private final Index index;
+    private final boolean[] info;
+
+    /**
+     * @param index of the person in the filtered person list to copy
+     * @param info list of details to copy to clipboard
+     */
+    public CopyCommand(Index index, boolean[] info) {
+        requireNonNull(index);
+
+        this.index = index;
+        this.info = info;
+    }
+
+    @Override
+    public String execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToCopy = lastShownList.get(index.getZeroBased());
+        String personDetails = copyToClipboard(personToCopy, info);
+
+        return String.format(Messages.MESSAGE_COPIED, personDetails);
+    }
+
+    /**
+     * Copies the details of {@code personToCopy} to clipboard
+     */
+    private static String copyToClipboard(Person personToCopy, boolean[] info) {
+        assert personToCopy != null;
```

Comment:
Should we use `requireNonNull()` instead?

Reply:
Uses `requireNonNull()` now